kratos/includes/user_provided_linear_master_slave_constraint.h

```diff
@@ -0,0 +1,319 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aditya Ghantasala
+//
+//
+
+#if !defined(USER_PROVIDED_LINEAR_MASTER_SLAVE_CONSTRAINT_H)
+#define USER_PROVIDED_LINEAR_MASTER_SLAVE_CONSTRAINT_H
+// System includes
+
+// project includes
+#include "includes/define.h"
+#include "includes/master_slave_constraint.h"
+
+namespace Kratos
+{
+/** \brief Constraint * @class MasterSlaveRelation
+    * @ingroup KratosCore
+    * @brief
+	* A class that implements the interface for different master-slave constraints to be applied on a system.
+    * This is the part that is seen by the user from the python level. Objects of this class are
+    * first class citizens of the modelpart.
+    *
+    * This class allows to add a master-slave constraint which is of the form
+    *
+    * SlaveDofVector = T * MasterDofVector + ConstantVector. (Processing of this is currently not implemented.)
+    *
+    * or
+    *
+    * SlaveDof = weight * MasterDof + Constant
+    *
+    * the data T and ConstantVector (or the equivalent scalars) are not stored in the base class, since they can be eventually evaluated on the flight
+    * 
+    * This class's object will provide its slave, master details and relation matrix between them.
+    *
+    * One can add two UserProvidedLinearMasterSlaveConstraint objects with same slave but different masters and weights.
+    * Consider user adds : SlaveDof = weight1 * MasterDof1 + Constant1
+    *              and   : SlaveDof = weight2 * MasterDof2 + Constant2
+    *
+    * These are later consolidated in the builder and solver to make
+    *                    : SlaveDof = weight1 * MasterDof1 + weight2 * MasterDof2 + Constant1+Constant2
+    *       and then converted to :
+    *                    : SlaveEqID = weight1 * MasterEqId1 + weight2 * MasterEqId2 + Constant1+Constant2
+    * This unique equation is used later on to modify the equation system.
+    */
+class UserProvidedLinearMasterSlaveConstraint :  public MasterSlaveConstraint
```

Comment:
This seems like a strange name to me ...
Why not `LinearMasterSlaveConstraint`?

Reply:
Ok for me to change it